src/index.js

```diff
@@ -1,26 +1,38 @@
 // @flow
 import React from "react";
 
+declare function __webpack_require__(pathOrId: string | number): any;
+declare var __webpack_modules__: [];
 type GenericComponent<Props> = Class<React.Component<{}, Props, mixed>>;
 type LoadedComponent<Props> = GenericComponent<Props>;
 type LoadingComponent = GenericComponent<{}>;
 
+let isWebpack = typeof __webpack_require__ !== "undefined";
+let requireFn = isWebpack ? __webpack_require__ : module.require;
+
 let babelInterop = obj => {
   // $FlowIgnore
   return obj && obj.__esModule ? obj.default : obj;
 };
 
 let tryRequire = (pathOrId: string | number) => {
   try {
+    if (
+      isWebpack &&
+      typeof pathOrId === "number" &&
+      !__webpack_modules__[pathOrId]
+    ) {
+      // if it's not in webpack modules, we wont be able
+      // to load it. If we attempt to, we mess up webpack's
+      // internal state, so we exit early
+      return null;
```

Comment:
Here's a bit of the `__webpack_require__` function:

```javascript

/******/ 	// The require function
/******/ 	function __webpack_require__(moduleId) {

/******/ 		// Check if module is in cache
/******/ 		if(installedModules[moduleId])
/******/ 			return installedModules[moduleId].exports;

/******/ 		// Create a new module (and put it into the cache)
/******/ 		var module = installedModules[moduleId] = {
/******/ 			i: moduleId,
/******/ 			l: false,
/******/ 			exports: {}
/******/ 		};

/******/ 		// Execute the module function
/******/ 		modules[moduleId].call(module.exports, module, module.exports, __webpack_require__);
// ... yada yada
```

So, from this we learn that `installedModules` gets initialized with the given `moduleId` when you call `__webpack_require` and then the next time you try to call `__webpack_require`, it exits early with what it is initialized with. So what was happening was we'd call it here, then my dynamic import would call it and it would exit early with that empty `export` object.

So we can protect against this by not calling the `requireFn` if `__webpack_modules__` (which I believe is just `installedModules` has not yet been initialized with the module).